paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsAdapter.kt

```diff
@@ -428,6 +448,145 @@ internal class PaymentOptionsAdapter(
                 )
                 else -> ""
             }
+
+            fun getRemoveDescription(resources: Resources) = resources.getString(
+                R.string.stripe_paymentsheet_remove_pm,
+                getDescription(resources)
+            )
+        }
+    }
+    internal companion object {
+        private fun calculateViewWidth(parent: ViewGroup): Dp {
+            val targetWidth = parent.measuredWidth - parent.paddingStart - parent.paddingEnd
+            val screenDensity = parent.context.resources.displayMetrics.density
+            // minimum width for each item, accounting for the CardView margin so that the CardView
+            // is at least 100dp wide
+            val minItemWidth = 100 * screenDensity + 6.0f + 6.0f
+            // numVisibleItems is incremented in steps of 0.5 items (1, 1.5, 2, 2.5, 3, ...)
+            val numVisibleItems = (targetWidth * 2 / minItemWidth).toInt() / 2f
+            val viewWidth = (targetWidth / numVisibleItems)
+            return (viewWidth / screenDensity).dp
+        }
+    }
+}
+
+@Composable
+fun PaymentOptionUi(
+    viewWidth: Dp,
+    isSelected: Boolean,
+    isEditing: Boolean,
+    isEnabled: Boolean,
+    iconRes: Int,
+    accessibilityDescription: String,
+    labelText: String? = "",
+    onRemoveListener: (() -> Unit)? = null,
+    onRemoveAccessibilityDescription: String = "",
+    onItemSelectedListener: (() -> Unit)
+) {
+    val strokeColor = colorResource(
+        if (isSelected) {
+            R.color.stripe_paymentsheet_payment_option_selected_stroke
+        } else {
+            R.color.stripe_paymentsheet_payment_option_unselected_stroke
+        }
+    )
+
+    val cardBackgroundColor = if (isEnabled) {
+        MaterialTheme.colors.surface
+    } else {
+        colorResource(id = R.color.stripe_paymentsheet_elements_background_disabled)
+    }
+
+    ConstraintLayout(
```

Comment:
nit: Could you add a comment on why ConstraintLayout is required, I think there are some optimization if you use Columns and Rows where possible.

Reply:
done